agents/ibootbar/ibootbar.py

```diff
@@ -0,0 +1,520 @@
+import numpy as np
+from numpy import random
+import datetime
+import time
+import os
+from os import environ
+
+from twisted.internet.defer import inlineCallbacks
+from autobahn.twisted.util import sleep as dsleep
+from twisted.internet import reactor
+import argparse
+import txaio
+from ocs.ocs_twisted import TimeoutLock
+
+from socs.snmp import SNMPTwister
+
+# For logging
+txaio.use_twisted()
+
+on_rtd = os.environ.get('READTHEDOCS') == 'True'
+if not on_rtd:
+    from ocs import ocs_agent, site_config
+
+
+class ibootbarSNMP:
```

Comment:
I'm glad the `MeinbergSNMP` class was useful here (though the whole timing structure I built for that to query in interval groups is perhaps not really needed here.) However, there's enough code re-use here that it'd be worth pulling a lot of this out to a class that `ibootbarSNMP` and `MeinbergSNMP` can use as a base class, for instance.